app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/inspections/AppEngineDeprecatedRuntimeInspection.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.java.inspections;
+
+import com.google.cloud.tools.intellij.appengine.java.AppEngineMessageBundle;
+import com.google.cloud.tools.intellij.appengine.java.util.AppEngineUtil;
+import com.google.common.collect.ImmutableList;
+import com.intellij.codeInspection.LocalQuickFix;
+import com.intellij.codeInspection.ProblemDescriptor;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.codeInspection.XmlSuppressableInspectionTool;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiElementVisitor;
+import com.intellij.psi.XmlElementVisitor;
+import com.intellij.psi.xml.XmlFile;
+import com.intellij.psi.xml.XmlTag;
+import java.util.List;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.Nls.Capitalization;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * An XML inspection that inspects appengine-web.xml files for deprecated App Engine Java runtime
+ * values. Deprecated values will be underlined with an error, and a quickfix to update to the
+ * supported Java 8 runtime will be presented
+ */
+public class AppEngineDeprecatedRuntimeInspection extends XmlSuppressableInspectionTool {
+
+  private List<String> deprecatedRuntimes = ImmutableList.of("java", "java7");
+
+  private static final String APP_ENGINE_JAVA8_RUNTIME_VALUE = "java8";
+  private static final String APP_ENGINE_WEB_XML_ROOT_TAG_NAME = "appengine-web-app";
```

Comment:
Do we have the same constant anywhere else or is it the first time we use it? It could probably be easier to use it once for all places if we do use it.